web/components/roadmap/MiniRoadmap.tsx

```diff
@@ -49,54 +53,93 @@ export const MiniRoadmap = (props: Props): ReactElement => {
-                <div className="fdr fac padding-left-05">
-                  <VerticalStepIndicator
-                    stepNumber={step.step_number}
-                    last={index === array.length - 1}
-                    isOpen={openSteps.includes(step.step_number)}
-                    active={step.step_number === activeStepNumber}
-                    small={true}
-                    completed={isStepCompleted(step, userData)}
-                    key={openSteps.join(",")}
-                  />
-                  <button
-                    className="usa-button--unstyled width-100"
-                    onClick={() => toggleStep(step.step_number)}
-                    aria-expanded={openSteps.includes(step.step_number)}
-                  >
-                    <div className=" step-label sm fdr fjc fac">
-                      <h3
-                        className={`margin-0 font-body-xs line-height-body-2 text-ink ${
-                          step.step_number === activeStepNumber ? "text-primary-dark" : "weight-unset"
-                        }`}
-                        data-step={step.step_number}
-                      >
-                        {step.name}
-                      </h3>
-                      <div className="mla fdc fac">
-                        <Icon className="usa-icon--size-3 text-ink">
-                          {openSteps.includes(step.step_number) ? "expand_less" : "expand_more"}
-                        </Icon>
-                      </div>
-                    </div>
-                  </button>
-                </div>
-                <div className="margin-left-5 font-sans-xs">
-                  {openSteps.includes(step.step_number) &&
-                    step.tasks.map((task) => (
-                      <MiniRoadmapTask
-                        key={task.id}
-                        task={task}
-                        active={task.id === props.activeTaskId}
-                        onTaskClick={props.onTaskClick}
-                      />
-                    ))}
-                </div>
-              </div>
+              <MiniRoadmapStep
+                step={step}
+                isLast={index === array.length - 1}
+                activeTaskId={props.activeTaskId}
+                completed={isStepCompleted(step, userData)}
+                isOpen={userData?.preferences.roadmapOpenSteps.includes(step.step_number)}
+                toggleStep={toggleStep}
+                onTaskClick={props.onTaskClick}
+                key={step.step_number}
+              />
             ))}
         </SectionAccordion>
       ))}
     </>
   );
 };
+
+interface StepProps {
+  step: Step;
+  isLast: boolean;
+  activeTaskId?: string;
+  isOpen?: boolean;
+  completed: boolean;
+  onTaskClick?: () => void;
+  toggleStep: (number: number, isOpen: boolean) => void;
+}
+
+export const MiniRoadmapStep = (props: StepProps): ReactElement => {
+  const [isOpen, setIsOpen] = useState<boolean>(props.isOpen ?? false);
```

Comment:
why do we track this independently and duplicate state here? could it be tracked from just whether the step is contained in openSteps? since the userdata is essentially state, wouldn't the update to the userData trickle downward to change the component, if the component were just figuring out for itself if it should be open. and the set-is-open-if-active functionality, instead of setting state, the render could use a function that computes whether the step is open. unless I am missing something